### PR TITLE
[IMP] website: redesign `s_process_steps` snippet

### PR DESCRIPTION
**Pull request by anso-odoo**
The goal of this PR is to improve the visual design of the default `s_process_steps` snippet.

| Before | After |
|--------|--------|
| Screenshot 2024-07-23 at 11 45 09 | Screenshot 2024-07-23 at 11 45 22 | 

Requires:
- https://github.com/odoo/design-themes/pull/840

task-3672430

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by anso-odoo** [comment]
@stefanorigano This one is ready for checking. Can you have a look ? Thx !

File: addons/website/views/snippets/s_process_steps.xml [hunk 1]
```diff
@@ -11,53 +11,54 @@
...
-                        <i class="fa fa-shopping-basket fa-2x mx-auto rounded-circle bg-o-color-1"/>
+                    <div class="s_process_step_number d-flex align-items-center justify-content-center mx-auto rounded-circle bg-primary-light">
+                        <h2 class="mb-0 text-primary text-center fw-normal">1</h2>
                     </div>
-                    <div class="s_process_step_content">
-                        <h2>Add to cart</h2>
-                        <p>Let your customers follow <br/>and understand your process.</p>
+                    <div class="s_process_step_content mt-3 px-3 text-center">
+                        <h3>Add to cart</h3>
+                        <p>Let your customers follow and understand your process.</p>
```
Comment (stefanorigano) [inline addons/website/views/snippets/s_process_steps.xml:25]: Can we reduce the length of this demo text? It easily goes in third line while the other are always 2 lines max

**Review by stefanorigano — CHANGES_REQUESTED** [review]
Thanks @anso-odoo , some remarks :)

File: addons/website/views/snippets/s_process_steps.xml [hunk 2]
```diff
@@ -11,53 +11,54 @@
...
-                <div class="col-lg-3 s_process_step pt24 pb24">
-                    <svg class="s_process_step_connector" viewBox="0 0 100 20" preserveAspectRatio="none">
+                <div class="col-lg-3 s_process_step position-relative pt24 pb24">
+                    <svg class="s_process_step_connector position-absolute z-index-1" viewBox="0 0 100 20" preserveAspectRatio="none">
                         <path d="M 0 10 L 100 10" vector-effect="non-scaling-stroke"/>
                     </svg>
-                    <div class="s_process_step_icon">
-                        <i class="fa fa-paypal fa-2x mx-auto rounded-circle bg-o-color-2"/>
+                    <div class="s_process_step_number d-flex align-items-center justify-content-center mx-auto rounded-circle bg-primary-light">
+                        <h2 class="mb-0 text-primary text-center fw-normal">3</h2>
```
[the remark this answers is not in the document]
Comment (anso-odoo) [reply addons/website/views/snippets/s_process_steps.xml:45]: I don't get what you mean sorry. It is still possible for them to toggle bold option this way, so it should be fine, no ?

File: addons/website/views/snippets/s_process_steps.xml [hunk 3]
```diff
@@ -82,6 +83,13 @@
...
                     data-css-property="stroke" data-change-color="true"/>
             </we-row>
+            <we-colorpicker
+                string="Step background"
+                data-apply-to=".s_process_step_number"
+                data-name="number_bg_opt"
+                data-select-style="true"
+                data-css-property="background-color"
+                data-color-prefix="bg-"/>
         </div>
```
[the remark this answers is not in the document]
Comment (anso-odoo) [reply addons/website/views/snippets/s_process_steps.xml]: Indeed it's better if we can change it per step. I don't know if we can easily inject the option in the column section of the sidebar but I'll have a look

Maybe we can deactivate the color preset option ? It's a bit "overkill" for an element with only 1 or 2 text characters ?

**Comment by anso-odoo** [comment]
Thanks for the review @stefanorigano , made some changes and added some comments !

File: addons/website/views/snippets/s_process_steps.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (stefanorigano) [reply addons/website/views/snippets/s_process_steps.xml:45]: Default is `500`.  `fw-normal` is `400`
By toggling [BOLD], the user will get `700`, not `500`... maining that, for this element only, he'll get stuck with either `400` or `700` and both are not consistent with the default font-weight.
For an user not happy with these values, the only option would be to remove the class by using an xpath.

Also,  the user could have set `$headings-font-weight: 200` in SCSS, and `fw-normal` would override his preferences 😕

Let me be clear, I do think that `400` is a better fit for **this** font-family and I think we should apply this value...  but globally and allowing the user to fine-tune it 😉 
This is one of those circumstances that highlights editor's limitations about font-weights in general... until these issue will be solved (see bold applied on `.lead`, for example...), I think that we should opt for consistency and predictability over fine-tuned visual results tailored for specific circumstances.

File: addons/website/views/snippets/s_process_steps.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (stefanorigano) [reply addons/website/views/snippets/s_process_steps.xml]: > Maybe we can deactivate the color preset option ?
👍

File: addons/website/views/snippets/s_process_steps.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (anso-odoo) [reply addons/website/views/snippets/s_process_steps.xml]: Ok i get it.. I guess we can let it by default indeed. It really depends on the font :)

File: addons/website/views/snippets/s_process_steps.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (anso-odoo) [reply addons/website/views/snippets/s_process_steps.xml]: I added the step background option in the column section. Can you check ? Thanks !

File: addons/website/static/src/snippets/s_process_steps/002.scss [hunk 4]
```diff
@@ -0,0 +1,53 @@
+.s_process_steps[data-vcss='002']  {
+    $process-step-number-size: 5rem;
+
+    .s_process_step {
+        .s_process_step_number {
+            width: $process-step-number-size;
+            height: $process-step-number-size;
+
+            // It is possible to add icons or images with the "/" command
+            img {
```
Comment (stefanorigano) [inline addons/website/static/src/snippets/s_process_steps/002.scss:10]: Nice 👍 

image

File: addons/website/views/snippets/s_process_steps.xml [hunk 5]
```diff
@@ -84,6 +85,17 @@
             </we-row>
         </div>
     </xpath>
+    <xpath expr="//t[@t-call='website.snippet_options_background_options']" position="after">
+        <t t-call="website.snippet_options_background_options">
+            <t t-set="background_label">Step Background</t>
+            <t t-set="selector" t-value="'.s_process_step'"/>
+            <t t-set="target" t-value="'.s_process_step_number'"/>
```
Comment (stefanorigano) [inline addons/website/views/snippets/s_process_steps.xml:92]: I'm not sure anymore, sorry. Placing the background controller into the "Column" entry streamlines the UI a bit, but it's not the expected location for the widget 😕 

I propose this :
```suggestion
            <t t-set="selector" t-value="'.s_process_step .s_process_step_number'"/>
```

File: addons/web_editor/views/snippets.xml [hunk 6]
```diff
@@ -112,7 +112,7 @@
          t-att-data-selector="selector"
          t-att-data-exclude="exclude"
          t-att-data-target="target">
-        <we-row t-if="with_colors or with_images" string="Background" class="o_we_full_row">
+        <we-row t-if="with_colors or with_images" t-att-string="background_label or 'Background'" class="o_we_full_row">
```
Comment (stefanorigano) [inline addons/web_editor/views/snippets.xml:115]: Not necessary anymore, read my previous comment.
Sorry for the hassle 🙏🏼

**Review by stefanorigano — CHANGES_REQUESTED** [review]

**Review by stefanorigano — APPROVED** [review]
Great, LGTM, ready for tech. review 👍 
Thanks!

**Comment by bso-odoo** [comment]
@robodoo delegate=loco-odoo

**Comment by loco-odoo** [comment]
@anso-odoo , just a small point: you put the "Our process in four easy steps" title in an `h3` heading element and the steps numbers into `h2` heading elements. Should we not swap them (put the title in an `h2` heading element and the steps number into `h3` heading elements)? Otherwise, it looks good to me, thanks for your PR :pray:

**Comment by anso-odoo** [comment]
Hello @loco-odoo, makes sense, i made the changes. Thanks !

**Comment by loco-odoo** [comment]
@robodoo r+

**Pull request closed without merging**